presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+                    e);
+        }
+
+        try {
+            // We'll then delete the metadata
+            metaManager.deleteTableMetadata(oldName);
+        }
+        catch (Exception e) {
+            try {
+                // catch all to rollback the operation -- rename the table back and create the old
+                // metadata
+                conn.tableOperations().rename(newTable.getFullTableName(),
+                        oldTable.getFullTableName());
+
+                if (oldTable.isIndexed()) {
+                    conn.tableOperations().rename(newTable.getIndexTableName(),
+                            oldTable.getIndexTableName());
+                    conn.tableOperations().rename(newTable.getMetricsTableName(),
+                            oldTable.getMetricsTableName());
+                }
+
+                metaManager.createTableMetadata(oldTable);
+            }
+            catch (Exception e1) {
+                throw new PrestoException(INTERNAL_ERROR,
+                        "Failed to rollback rename table operation ", e1);
+            }
+
+            throw new PrestoException(INTERNAL_ERROR, "Failed to delete metadata",
+                    e);
+        }
+    }
+
+    /**
+     * Creates a new view with the given name and data.  Replaces a view if 'replace' is true.
+     *
+     * @param viewName Name of the view
+     * @param viewData Data for the view
+     * @param replace True to replace, false otherwise
+     */
+    public void createView(SchemaTableName viewName, String viewData, boolean replace)
+    {
+        if (!replace && this.getSchemaNames().contains(viewName.getSchemaName())) {
+            if (this.getViewNames(viewName.getSchemaName()).contains(viewName.getTableName())) {
+                throw new PrestoException(ALREADY_EXISTS, "View already exists");
+            }
+
+            if (this.getTableNames(viewName.getSchemaName()).contains(viewName.getTableName())) {
+                throw new PrestoException(ALREADY_EXISTS, "View already exists as data table");
+            }
+        }
+
+        if (replace && getView(viewName) != null) {
+            metaManager.deleteViewMetadata(viewName);
+        }
+
+        metaManager.createViewMetadata(new AccumuloView(viewName.getSchemaName(), viewName.getTableName(), viewData));
+    }
+
+    /**
+     * Drops the given view
+     *
+     * @param viewName View to drop
+     */
+    public void dropView(SchemaTableName viewName)
+    {
+        metaManager.deleteViewMetadata(viewName);
+    }
+
+    /**
+     * Adds a new column to an existing table
+     *
+     * @param table Accumulo table to act on
+     * @param column New column to add
+     * @throws IndexOutOfBoundsException If the ordinal is greater than or equal to the number of columns
+     */
+    public void addColumn(AccumuloTable table, AccumuloColumnHandle column)
+            throws IndexOutOfBoundsException
+    {
+        // Add the column
```

Comment:
Aren't there any constraints for Accumulo to not have duplicate columns? I don't see any check for this here.